storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -344,12 +347,29 @@ private void downloadDistributeStormCode(Map conf, String stormId, String master
 
     /**
      * Assert if all blobs are downloaded for the given topology
-     * 
+     *
+     * @param targetDir
+     * @param blobFileNames
+     * @return
+     */
+    protected boolean didDownloadBlobsForTopologySucceed(String targetDir, List<String> blobFileNames) throws IOException {
+        for (String string : blobFileNames) {
+            if (!Utils.checkFileExists(targetDir, string)) {
+                LOG.info("Fail to find downloaded file: dir {} filename {}", targetDir, string);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Assert if all blobs in blobstore map are downloaded for the given topology
+     *
      * @param stormconfPath
      * @param targetDir
      * @return
      */
-    protected boolean didDownloadBlobsForTopologySucceed(String stormconfPath, String targetDir) throws IOException {
+    protected boolean didDownloadBlobsInBlobStoreMapForTopologySucceed(String stormconfPath, String targetDir) throws IOException {
```

Comment:
Same as above, `areBlobsDownloaded` would be fine for me since we're commenting like that.